### PR TITLE
Improve the documentation

### PR DESCRIPTION
**Pull request by pyrech**
The documentation have been reworked a bit:
- Smaller main readme with a "Quick start" chapter
- Move documentation to `/docs` directory to match GitHub standards
- Better structure and navigation in the documentation
- Add documentation about iteration / pagination
- Add a GitHub pages for the whole documentation with [Docsify](https://docsify.js.org):
  - only a small index.html is commited in `docs/` folder to configure Docsify
  - nothing to build - it automatically transforms existing Markdown file from the `main` branch in HTML page

You can see the final result here: https://jolicode.github.io/slack-php-api/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,22 @@
-# A PHP client for Slack's API
+# Introduction
```
Comment (damienalexandre) [inline README.md:1]: "Introduction" as the main Readme title is kind of ill. The previous title was better, is that a docsify issue?

File: README.md [hunk 2]
```diff
@@ -28,35 +31,16 @@ You can now install the Slack client:
...
+$client = JoliCode\Slack\ClientFactory::create($yourSlackToken);
 
 $user = $client->usersInfo(['user' => 'U123AZER'])->getUser();
-
-dump($user);
 ```
 
-<p align="center">
-  <img src="doc/images/model-sample.png" alt="Sample user object" />
-<p>
```
Comment (damienalexandre) [inline README.md:51]: Having some way to display the result in the quick start was great. Maybe we could show a "var_dump" output without image?

File: docs/1-get-started.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+You can now install the Slack client:
+
+```bash
+composer require jolicode/slack-php-api
+```
+
+## Quick start
+
+```php
```
Comment (damienalexandre) [inline docs/1-get-started.md:24]: Before that snippet, a link to the "get your token" is needed. There is no "quick start" without token IMO.

File: docs/1-get-started.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// $client contains all the methods to interact with the API
+$client = JoliCode\Slack\ClientFactory::create($yourSlackToken);
+
+$user = $client->usersInfo(['user' => 'U123AZER'])->getUser();
+```
+
```
Comment (damienalexandre) [inline docs/1-get-started.md:30]: Could you add a sentence here about what we get in return from this code?

File: docs/2-usage.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
+# Usage
+
+## Slack token
+
+Before you can use this client, you need to retrieve a token from Slack.
+
+Checkout Slack's documentation about [all different kind of tokens](https://api.slack.com/authentication/token-types).
+A good starting point is the [Authentication Basics documentation](https://api.slack.com/authentication/basics).
```
Comment (damienalexandre) [inline docs/2-usage.md:8]: That part could be in the quick start.

File: docs/2-usage.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+```php
+<?php
+//...
+
+$members = $client->usersList(['limit' => 100])->getMembers();
+```
+
+## Iterating over pagination
+
+Some Slack API endpoints use a token based pagination.
```
Comment (damienalexandre) [inline docs/2-usage.md:36]:
```suggestion
Some Slack API endpoints use a cursor based pagination.
```

File: docs/3-troubleshoots.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+miss some features.
+
+If you miss an option in a method, until Slack add it to their spec, we can
+manually add it to our [versioned spec](https://github.com/jolicode/slack-php-api/tree/main/resources/slack-openapi-patched.json) then
+[regenerate](updating-sdk.md) and release a new version of the library.
+
+## Missing data in the DTO?
+
+The Slack specification is not up to date and miss some critical parts. We do
+build a [better one on top of the official](doc/updating-sdk.md) but it can't
```
Comment (damienalexandre) [inline docs/3-troubleshoots.md:18]: Wrong file path?

```suggestion
build a [better one on top of the official](4-updating-sdk.md) but it can't
```

**Review by damienalexandre — COMMENTED** [review]
Love the website, it makes the documentation easier to read. A link to it from the main README is missing I think.

Thanks a lot!

File: README.md (hunk 1 above)
Comment (pyrech) [reply README.md:1]: Nope, I started by including the readme directly in the documentation. But in the end I no longer use it, so I will restore this title :+1:

File: README.md (hunk 2 above)
Comment (pyrech) [reply README.md:51]: I don't find this really useful and rather too big in the main readme. I expect someone using the Quick start only to explore by itself the code with its favorite editor

File: docs/1-get-started.md (hunk 3 above)
Comment (pyrech) [reply docs/1-get-started.md:24]: Okay, I will move the token chapter here :+1:

**Review by damienalexandre — APPROVED** [review]

**Pull request merged**